src/binary/merkle_tree.rs

```diff
@@ -69,18 +69,18 @@ where
         Ok(tree)
     }
 
-    pub fn root(&mut self) -> Result<Bytes32, MerkleTreeError<StorageError>> {
-        let root_node = self.root_node()?;
+    pub fn root(&self) -> Bytes32 {
+        let mut scratch_storage = StorageMap::<NodesTable>::new();
+        let root_node = self.root_node(&mut scratch_storage);
         let root = match root_node {
             None => *empty_sum(),
             Some(ref node) => *node.hash(),
         };
-
-        Ok(root)
+        root
```

Comment:
```suggestion
         match root_node {
            None => *empty_sum(),
            Some(ref node) => *node.hash(),
        }
```

You can return from the match and avoid the assign-return